### PR TITLE
Fix/nico/auth

### LINKED ISSUE #31: Bug: submitting tags and url
getting error in middleware/auth.ts . on etheres.transaction.from()
TypeError: Cannot read properties of undefined (reading 'Transaction')
    at verifySignedFunctionMessage (/Projects/channel4/match-making/src/middleware/auth.ts:48:27)

### PR DESCRIPTION
**Pull request by NicoSerranoP**
Close #31

1. Fix `.env` not read by npm problem
2. `typechain` contract type compilation implemented (not required to fix the bug but I was working on it)
3. Use the new contract ABI (latest version)
4. Disable metaTx auth until WalletConnect/web3auth is implemented in frontend. We will trust the frontend and the users that they are sending urls from their wallet and not trying putting non-authorized addresses in the request to create URLs, Tags or Likes
5. **Fix url creation bug**

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import * as dotenv from "dotenv";
-dotenv.config({ path: __dirname+'/.env' });
+dotenv.config();
```
Comment (hackertron) [inline src/server.ts]: No Nico , the env is failing if we don't specify the path. Had this issue on render too and facing it locally too. 
```
dotenv.config({ path: __dirname+'/.env' });
```
using dotenv like above make sure it works.

**Review by hackertron — CHANGES_REQUESTED** [review]

File: src/server.ts (hunk 1 above)
Comment (NicoSerranoP) [reply src/server.ts]: That is weird :S are you using Linux? In my Windows setup it fails when we specify the path :(

Comment (NicoSerranoP) [reply src/server.ts]: I just pushed a new change reverting this, could you try to run this in your local? :)

Comment (hackertron) [reply src/server.ts]: Yes I am running Linux . I can put it in try catch block i guess.

Comment (NicoSerranoP) [reply src/server.ts]: I will read more about it and try to fix it in https://github.com/State-Channel-4/backend/tree/feat/nico/sync-contract

**Review by hackertron — APPROVED** [review]

**Pull request merged**